### ISSUE TITLE
jupyter2: red errors (e.g. "timeout") should clear upon each successful cell eval

### ISSUE THREAD

**Issue by haraldschilly** [sagemathinc/cocalc#3557]
I can evaluate cells fine, but there is a "timeout" warning at the top and it doesn't clear on its own.
Hence my thought is to add a line to the code where a cell is run to clear up "timeout" when evaluating cells clearly works. (other UI errors also clear out if the document in question works fine, e.g. compiling latex...)

![screenshot from 2019-02-09 09-23-19](https://user-images.githubusercontent.com/207405/52518573-b8bc5e00-2c4c-11e9-8f1d-513a1282b3f9.png)

**Comment by haraldschilly** [comment]
it's any error, e.g. this one is also sticky while the notebook works fine

![screenshot from 2019-02-09 10-07-47](https://user-images.githubusercontent.com/207405/52518894-a6452300-2c52-11e9-801a-3f5bb304c61b.png)

**Comment by williamstein** [comment]
1. For an error like "timeout" the bug is that the error is not descriptive enough.  What timed out?  That needs to be fixed.   The second screenshot above is the result of me trying to improve these error messages.

2. I disagree with " red errors (e.g. "timeout") should clear upon each successful cell eval" in complete generality.  What if the error has nothing to do with cell evaluation (e.g., maybe it is an error saving, but cell evaluation is working fine).  In that case, we shouldn't clear the error.    That said, in the case you posted above about getting kernel info, there is no point in showing that error once kernel info has been successfully obtained.  We could make it so the error message has a tag, and certain events could trigger clearing any visible errors with that that tag.   For this, of course, we would also want to make it possible to show more than one error at once, which is currently not possible (I was just too lazy to implement that).

3. In most cases if there is a timeout error, we should look deeper and ensure that whatever causes the timeout doesn't happen, or that the UI better communicates what is going on...

**Comment by haraldschilly** [comment]
Well, since this came up again, I elevated the priority. This ticket is mainly about that first screenshot. I don't know what times out, my best guess is evaluating a cell. 

To me, this happens regularly when I open up a notebook that was in the background for a while (maybe project did stop?). I can try to find a way to reproduce it, although just from the top of my head I don't know a way. I can only say that I see it quite often.

**Comment by williamstein** [comment]
I think the right first step is make it so any possible "timeout" (or other) error message explains itself better.  Then we'll know what causes that stupid timeout, and fix it.  I should do this.

**Comment by haraldschilly** [comment]
I found a way to reproduce it (this means, that banner shows up twice in a row)

1. python3.ipynb, and there is a cell `import numpy as np`, i.e. nothing special
2. goto project settings, click on stop to stop it and wait until it says stopped
3. open python3.ipynb tab and run that cell
4. immediately switch away to another project (I think this is necessary, because I tried it once without tabbing away and that timeout banner didn't show up)
5. after ~30 secs go back to the notebook
6. running the same cell again works fine, it starts with the counter `[1]`.
